test/infrastructure/docker/Dockerfile

```diff
@@ -18,14 +18,16 @@ FROM golang:1.13.5 as builder
 ARG goproxy=https://proxy.golang.org
 ENV GOPROXY=$goproxy
 
-WORKDIR /workspace/test/infrastructure/docker
-COPY go.mod go.mod
-COPY go.sum go.sum
-RUN go mod download
+# This needs to build with the entire Cluster API context
 WORKDIR /workspace
+# Copy in cluster-api (which includes the test/infrastructure/docker subdirectory)
 COPY . .
 
-# Build
+# Essentially, change directories into CAPD
+WORKDIR /workspace/test/infrastructure/docker
+RUN go mod download
```

Comment:
nit: Any changes to any file in the repo cause the `COPY . .` layer above to be invalidated in the cache. Splitting the go module download into a separate layer here won't help with caching or speed up builds, unfortunately. You can keep this if you want, but I'd probably remove it.

Reply:
yeah, good catch, removing